File: ChatGPT(예진)/Cpp (수정 완료)/Finding_the_Street_Between_Stores.cpp
#include <iostream>
#include <vector>
#include <cmath>
#include <limits>

using namespace std;

// Store class definition
class Store {
public:
    string name;
    float x;
    float y;

    Store(string n, float a, float b) : name(n), x(a), y(b) {}
};

// Calculate the Euclidean distance between two points
float euclidean_distance(const Store& point1, const Store& point2) {
    return sqrt(pow((point1.x - point2.x), 2) + pow((point1.y - point2.y), 2));
}

// Find the closest and farthest two stores
void find_closest_and_farthest(const vector<Store>& stores, vector<string>& closest_stores, float& min_distance, vector<string>& farthest_stores, float& max_distance) {
    min_distance = numeric_limits<float>::infinity();
    max_distance = -numeric_limits<float>::infinity();

    for (size_t i = 0; i < stores.size(); i++) {
        for (size_t j = i + 1; j < stores.size(); j++) {
            float distance = euclidean_distance(stores[i], stores[j]);
            if (distance < min_distance) {
                min_distance = distance;
                closest_stores[0] = stores[i].name;
                closest_stores[1] = stores[j].name;
            }
            if (distance > max_distance) {
                max_distance = distance;
                farthest_stores[0] = stores[i].name;
                farthest_stores[1] = stores[j].name;
            }
        }
    }
}

// Main function
int main() {
    vector<Store> stores;
    vector<string> closest_stores(2);
    vector<string> farthest_stores(2);
    float min_distance, max_distance;

    cout << "Enter the locations of three stores." << endl;
    for (int i = 0; i < 3; i++) {
        string store_name;
        float x, y;
        cout << "Enter the store name: ";
        cin >> store_name;
        cout << "Enter the x coordinate of the store: ";
        cin >> x;
        cout << "Enter the y coordinate of the store: ";
        cin >> y;
        stores.push_back(Store(store_name, x, y));
    }

    find_closest_and_farthest(stores, closest_stores, min_distance, farthest_stores, max_distance);
    cout << "Closest two stores: " << closest_stores[0] << ", " << closest_stores[1] << " Distance: " << min_distance << endl;
    cout << "Farthest two stores: " << farthest_stores[0] << ", " << farthest_stores[1] << " Distance: " << max_distance << endl;

    return 0;
}

File: ChatGPT(예진)/Java (수정 완료)/Finding_the_Street_Between_Stores.java
import java.util.Scanner;
import java.lang.Math;

class Store {
    String name;
    float x;
    float y;
}

public class Finding_the_Street_Between_Stores {

    // Function to input store locations
    public static void inputStoreLocations(Store[] stores) {
        Scanner scanner = new Scanner(System.in);
        for (int i = 0; i < 3; i++) {
            stores[i] = new Store();
            System.out.print("Enter the store name: ");
            stores[i].name = scanner.nextLine();
            System.out.print("Enter the x-coordinate of the store: ");
            stores[i].x = scanner.nextFloat();
            System.out.print("Enter the y-coordinate of the store: ");
            stores[i].y = scanner.nextFloat();
            scanner.nextLine(); // consume newline
        }
    }

    // Function to calculate the Euclidean distance between two points
    public static float euclideanDistance(Store point1, Store point2) {
        return (float) Math.sqrt(Math.pow((point1.x - point2.x), 2) + Math.pow((point1.y - point2.y), 2));
    }

    // Function to find the closest and farthest two stores
    public static void findClosestAndFarthest(Store[] stores, String[] closestStores, float[] minDistance, String[] farthestStores, float[] maxDistance) {
        minDistance[0] = Float.POSITIVE_INFINITY;
        maxDistance[0] = Float.NEGATIVE_INFINITY;

        for (int i = 0; i < 3; i++) {
            for (int j = i + 1; j < 3; j++) {
                float distance = euclideanDistance(stores[i], stores[j]);
                if (distance < minDistance[0]) {
                    minDistance[0] = distance;
                    closestStores[0] = stores[i].name;
                    closestStores[1] = stores[j].name;
                }
                if (distance > maxDistance[0]) {
                    maxDistance[0] = distance;
                    farthestStores[0] = stores[i].name;
                    farthestStores[1] = stores[j].name;
                }
            }
        }
    }

    public static void main(String[] args) {
        Store[] stores = new Store[3];
        String[] closestStores = new String[2];
        String[] farthestStores = new String[2];
        float[] minDistance = new float[1];
        float[] maxDistance = new float[1];

        System.out.println("Enter the locations of three stores.");
        inputStoreLocations(stores);
        findClosestAndFarthest(stores, closestStores, minDistance, farthestStores, maxDistance);
        System.out.printf("The closest two stores are: %s, %s Distance: %.2f\n", closestStores[0], closestStores[1], minDistance[0]);
        System.out.printf("The farthest two stores are: %s, %s Distance: %.2f\n", farthestStores[0], farthestStores[1], maxDistance[0]);
    }
}
